src/components/PopoverButton/Popover.js

```diff
@@ -0,0 +1,117 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { Manager, Reference, Popper } from 'react-popper';
+import styled from 'react-emotion';
+
+const modifiers = {
+  offset: { enabled: true, offset: '0,10' },
+  flip: { enabled: true }
+};
+
+class Popover extends Component {
+  static propTypes = {
+    /**
+     * function rendering the popover
+     */
+    renderPopover: PropTypes.func.isRequired,
+    /**
+     * function rendering the reference (button or something clickable)
+     */
+    renderReference: PropTypes.func.isRequired,
+    /**
+     * placement of the popover relative to the button
+     */
+    placement: PropTypes.oneOf(['top', 'bottom', 'left', 'right']),
+    /**
+     * alignment of the popover relative to the button
+     */
+    align: PropTypes.oneOf(['start', 'end', 'center'])
+  };
+
+  static defaultProps = {
+    placement: 'bottom',
+    align: 'start'
+  };
+
+  state = { isOpen: false };
+
+  componentDidMount() {
+    document.addEventListener('click', this.onDocumentClick, false);
```

Comment:
Should these listeners maybe only be registered and unregistered when the popover is toggled? They are only relevant when it's open.